[CI] RareTermsIT testSingleValuedString failing

it seems to be related to the minimum version supported by the cluster:

```
»  Caused by: org.elasticsearch.action.search.VersionMismatchException: One of the shards is incompatible with the required minimum version [8.0.0]
»  	at org.elasticsearch.action.search.AbstractSearchAsyncAction.getConnection(AbstractSearchAsyncAction.java:701) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.AbstractSearchAsyncAction.lambda$checkMinimumVersion$0(AbstractSearchAsyncAction.java:243) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:?]
»  	at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602) ~[?:?]
»  	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
»  	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502) ~[?:?]
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/oytjgm34qzzps/tests/:qa:mixed-cluster:v7.15.0%23mixedClusterTest/org.elasticsearch.backwards.RareTermsIT/testSingleValuedString

**Reproduction line:**
`./gradlew ':qa:mixed-cluster:v7.15.0#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.RareTermsIT" -Dtests.method="testSingleValuedString" -Dtests.seed=3A3BEC681332358F -Dtests.bwc=true -Dtests.locale=el-CY -Dtests.timezone=Asia/Riyadh -Druntime.java=17`

**Applicable branches:**
master, 7.x

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.backwards.RareTermsIT&tests.test=testSingleValuedString

**Failure excerpt:**
```
org.elasticsearch.client.ResponseException: method [POST], host [http://127.0.0.1:45633], URI [idx/_search], status line [HTTP/1.1 500 Internal Server Error]
{"error":{"root_cause":[],"type":"search_phase_execution_exception","reason":"","phase":"fetch","grouped":true,"failed_shards":[],"caused_by":{"type":"illegal_state_exception","reason":"already closed"}},"status":500}

  at __randomizedtesting.SeedInfo.seed([3A3BEC681332358F:ACA67E8C28E2455E]:0)
  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:330)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:296)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:270)
  at org.elasticsearch.backwards.RareTermsIT.assertNumRareTerms(RareTermsIT.java:66)
  at org.elasticsearch.backwards.RareTermsIT.testSingleValuedString(RareTermsIT.java:56)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

Actually it seems a legit bug:

```
»  Caused by: java.lang.IllegalStateException: already closed
»  	at org.elasticsearch.common.breaker.PreallocatedCircuitBreakerService$PreallocatedCircuitBreaker.addWithoutBreaking(PreallocatedCircuitBreakerService.java:131) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.AggregatorBase.addRequestCircuitBreakerBytes(AggregatorBase.java:136) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.bucket.terms.AbstractRareTermsAggregator.lambda$newFilter$0(AbstractRareTermsAggregator.java:68) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.SetBackedScalingCuckooFilter.convert(SetBackedScalingCuckooFilter.java:282) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.SetBackedScalingCuckooFilter.merge(SetBackedScalingCuckooFilter.java:331) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.bucket.terms.InternalMappedRareTerms.reduce(InternalMappedRareTerms.java:112) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.InternalAggregations.reduce(InternalAggregations.java:248) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:191) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:480) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:468) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:131) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:98) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:36) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:84) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
```

The issue is that we are reusing the 'SetBackedScalingCuckooFilter' while merging and this filter contains a breaker that is already closed.

--- a/qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/RareTermsIT.java
+++ b/qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/RareTermsIT.java
@@ -38,7 +38,6 @@ public class RareTermsIT extends ESRestTestCase {
         return id;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/74985")
     public void testSingleValuedString() throws Exception {
         final Settings.Builder settings = Settings.builder()
             .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 2)

